src/components/events/ExternalOpportunitiesModal.js

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import ComputerWindow from '../general/ComputerWindowComponent';
+import ExternalOpportunityCard from './ExternalOpportunityCard';
+import styles from '@/styles/components/ExternalOpportunitiesModal.module.css';
+
+const colorOptions = {
+  'wcs-pink': 'var(--wcs-pink)',
+  'wcs-blue': 'var(--wcs-blue)',
+  'wcs-pink-light': 'var(--light-pink)',
+  'wcs-blue-light': 'var(--light-blue)',
+  'wcs-pink-light-light': 'var(--light-light-pink)',
+  'wcs-blue-light-light': 'var(--light-light-blue)',
+};
+
+function ExternalOpportunitiesModalContent({
+  color = 'wcs-pink',
+  category,
+  events,
+  closeModal,
+}) {
+  return (
+    <div className={styles.container}>
+      <ComputerWindow
+        className={`${styles.window} ${styles[`window-${color}-light-light`]}`}
+        topbarColor={colorOptions[color]}
+      >
+        <div
+          className={`${styles.headingBlock} ${styles[`headingBlock-${color}-light`]}`}
+        >
+          <h2 className={styles.heading}>{category}</h2>
+        </div>
+        <div className={styles.eventsCards}>
+          {events.map((event, index) => (
+            <ExternalOpportunityCard
+              key={index}
+              color={color}
+              icon={event.icon}
+              title={event.title}
+              link={event.link}
+              date={event.date}
+              time={event.time}
+              location={event.location}
+              description={event.description}
+            />
+          ))}
+        </div>
+        <button
+          type="button"
+          onClick={closeModal}
+          className={`${styles.closeButton} ${styles[`closeButton-${color}-light`]}`}
+        >
+          Close
+        </button>
+      </ComputerWindow>
+    </div>
+  );
+}
+
+export default function ExternalOpportunitiesModal({
+  color,
+  category,
+  data,
+  isOpen,
+  closeModal,
+}) {
+  return isOpen ? (
+    <ExternalOpportunitiesModalContent
```

Comment:
Do you need this 2 layered component here? Could it be condensed into one?

Reply:
Pushed changes to be make it a single component! Though I originally referenced the pre-existing officer modal structure to maintain consistency:
```javascript
export default function OfficerModal({ isOpen, closeModal, officer }) {
  return isOpen ? OfficerInformation(closeModal, officer) : null;
}
```
Just wanted to point it out in case we wanted to refactor the other pages as well.

Reply:
I see yeah thats code from last year, we'll probably be changing up that code later anyways, but thanks for flagging it.